src/shared/clients/ecsClient.ts

```diff
@@ -5,70 +5,69 @@
 
 import { ECS } from 'aws-sdk'
 import { ext } from '../extensionGlobals'
+import { getLogger } from '../logger'
 import { ClassToInterfaceType } from '../utilities/tsUtils'
 
 export type EcsClient = ClassToInterfaceType<DefaultEcsClient>
+
+export type ServicesAndToken =  {
+    services: ECS.Service[],
+    nextToken?: string
+}
+
+export type ClustersAndToken = {
+    clusters: ECS.Cluster[],
+    nextToken?: string
+}
+
+const MAX_RESULTS = 50
 export class DefaultEcsClient {
     public constructor(public readonly regionCode: string) {}
 
-    public async *listClusters(): AsyncIterableIterator<string> {
+    public async listClusters(nextToken?: string): Promise<ClustersAndToken> {
         const sdkClient = await this.createSdkClient()
-        const request: ECS.ListClustersRequest = {}
-        do {
-            const response = await this.invokeListClusters(request, sdkClient)
-            if (response.clusterArns) {
-                yield* response.clusterArns
+        const clusterArnList = await sdkClient.listClusters({maxResults: MAX_RESULTS, nextToken}).promise()
```

Comment:
This PR was made before Husky was fixed so I see some style issues. I think if you make a small change to each file then do a commit it'll run prettier again.